src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/AuditUtility.cs

```diff
@@ -298,7 +299,8 @@ private static (string severityLabel, NuGetLogCode code) GetSeverityLabelAndCode
                                 affectedGraphs.Add(graph.TargetGraphName);
                             }
 
-                            if (!auditInfo.IsDirect && resolvedDependency.Parent == currentProject)
+                            if (!auditInfo.IsDirect &&
```

Comment:
I assume you mean on the following line, not the quoted line?

I don't know. Creating a hash set has a cost, and I expect most people will want to avoid using packages with vulnerabilities in their projects, so I don't expect this code path to be used very frequently. It would also be interesting to know what the 95th percentile is for the number of direct package references in a project is. If the number is small, then I suspect there will be little benefit to a hashset.